### ISSUE TITLE
Use of `aria-haspopup`

### ISSUE THREAD

**Issue by Firestorm980** [10up/component-navigation#20]
**Describe the bug**
Curious if the `aria-haspopup` attribute is the correct use of the attribute in the component. According to https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup, it should:

>Indicates the availability and type of interactive popup element, such as menu or dialog, that can be triggered by an element.
>
>A popup element usually appears as a block of content that is on top of other content. Authors MUST ensure that the role of the element that serves as the container for the popup content is menu, listbox, tree, grid, or dialog, and that the value of aria-haspopup matches the role of the popup container.

In https://inclusive-components.design/menus-menu-buttons/, it is specifically omitted in a _navigational_ menu but is entirely appropriate (and required) in an _application_ menu (using the `menu` role(s)).

On a related note, use of `aria-expanded` should be included.

**To Reproduce**
1. Clone repo
2. Open demo page locally
3. Open inspector or screen reader and note that links with submenus have `aria-haspopup`.

**Expected behavior**
Links with submenus shouldn't have `aria-haspopup`

**Additional context**
There seems to be conflicting information about this very issue, which probably has to do with the proper use of `menu` and its related roles.
